x/wasm-storage/keeper/filter.go

```diff
@@ -20,12 +26,27 @@ func ApplyFilter(filter []byte, reveals []RevealBody) ([]int, bool, error) {
 	}
 
 	switch filter[0] {
-	case filterNone:
+	case filterAlgoNone:
 		return make([]int, len(reveals)), true, nil
 
-	// TODO: Reactivate mode filter
-	// case filterMode:
-	// 	return nil, false, errors.New("filter type mode is not implemented")
+	case filterAlgoMode:
+		dataPath, err := types.UnpackModeFilter(filter)
+		if err != nil {
+			return nil, false, err
+		}
+
+		if dataPath == "" {
+			return nil, false, errors.New("empty JSON path in filter input [Mode]")
+		}
+
+		exitCodes := make([]uint8, 0, len(reveals))
+		revealData := make([]string, 0, len(reveals))
+		for _, r := range reveals {
+			exitCodes = append(exitCodes, r.ExitCode)
+			revealData = append(revealData, r.Reveal)
+		}
```

Comment:
This seems unnecessary to me. I think we should just feed `reveals` directly to `filterMode()`.

Reply:
here [8c49c59](https://github.com/sedaprotocol/seda-chain/pull/290/commits/8c49c59ff72231ea958fb1dd5e456622d9f10163)